### ISSUE TITLE
Set the right value for Speculation_Store_Bypass in proc/<pid>/status

### ISSUE THREAD

**Issue by sathnaga** [linuxppc/issues#255]
Currently powerpc system shows `unknown` default value for `Speculation_Store_Bypass` in `proc/<pid>/status`, it has to set either one of the following valid values,
`vulnerable`, `globally mitigated`, `thread vulnerable`, `thread mitigated`, `not vulnerable`

```
# cat /proc/self/status|grep Speculation_Store_Bypass:
Speculation_Store_Bypass:    unknown

# uname -r
5.2.0-rc6-00015-g249155c20f9b
# lsmcode|grep skiboot
 Product Extra         :     skiboot-v6.3-rc2

HW: Power9 DD2.2
```
 
Other arch implementations:
```
0 ssbd.c  120 int arch_prctl_spec_ctrl_get(struct task_struct *task, unsigned long which) -- arm
1 bugs.c 1035 int arch_prctl_spec_ctrl_get(struct task_struct *task, unsigned long which)   --x86
2 sys.c  2261 int __weak arch_prctl_spec_ctrl_get(struct task_struct *t, unsigned long which) -- generic
```

**Comment by mpe** [comment]
The value needs to be tied to both `stf_enabled_flush_types` and `stf_barrier`.

I think the logic is:
```
if stf_enabled_flush_types == STF_BARRIER_NONE:
  "not vulnerable"
else if stf_barrier:
  "globally mitigated"
else
  "vulnerable"
```

**Comment by walbon** [comment]
I'm wondering that:
```c
        if (stf_barrier) {
                if (stf_enabled_flush_types == STF_BARRIER_NONE)
                        return PR_SPEC_NOT_AFFECTED; // "not vulnerable"
                else
                        return PR_SPEC_DISABLE; // "globally mitigated"
        } else
                return PR_SPEC_DISABLE_NOEXEC; // ??? "vulnerable"

        return -EINVAL;
```

**Comment by walbon** [comment]
With patch the result is below.

```console
[root@ltc-zz4-lp14 ~]# uname -a
Linux ltc-zz4-lp14.aus.stglabs.ibm.com 5.3.0-rc1-issue255+ #2 SMP Tue Aug 6 21:55:39 EDT 2019 ppc64le ppc64le ppc64le GNU/Linux
[root@ltc-zz4-lp14 ~]# grep -i bypass /proc/23/status
Speculation_Store_Bypass:       globally mitigated
[root@ltc-zz4-lp14 ~]# cat /sys/devices/system/cpu/vulnerabilities/meltdown
Mitigation: RFI Flush, L1D private per thread
```

**Comment by walbon** [comment]
@mpe Do I must to check the status of RFI flush or another feature which could be disable via /sys then change the status for 'vulnerable' or 'mitigated'?
eg.:  echo 0 > /sys/kernel/debug/powerpc/rfi_flush

**Comment by mpe** [comment]
It's `spec_store_bypass`.

**Comment by mpe** [comment]
I think my original suggestion on this was wrong.

Our mitigation is not global, ie. it's a barrier that must be used in the right location. We have kernel code to insert the barrier on kernel entry/exit, but that doesn't protect userspace against itself (ie. sandboxes).

There's no way to express that with the current values as far as I can see.

I think all we can do for now is:
```
if stf_enabled_flush_types == STF_BARRIER_NONE:
  return PR_SPEC_NOT_AFFECTED // "not vulnerable"
else
  return PR_SPEC_ENABLE // "vulnerable"
```

To express the situation properly we'd need another value, something like `PR_SPEC_MITIGATION_AVAILABLE` (??) which says that there is a mitigation available but it must be used. That still has the problem that it doesn't tell userspace *what* the mitigation is, userspace would have to know.

**Comment by mpe** [comment]
I posted a v2:
https://patchwork.ozlabs.org/patch/1265576/

**Comment by mpe** [comment]
Merged as torvalds/linux@d93e5e2d03d4f41dfedb92200a2c0413ab8ee4e7.

Thanks @walbon for wading into this swamp :crocodile: and trying to implement my incorrect suggestions :medal_military:
